frontend/ui/widget/opdsbrowser.lua

```diff
@@ -474,6 +474,8 @@ function OPDSBrowser:downloadFile(item, format, remote_url)
     -- download to user selected directory or last opened dir
     local lastdir = G_reader_settings:readSetting("lastdir")
     local download_dir = G_reader_settings:readSetting("download_dir") or lastdir
+    item.author = item.author:gsub('[\\,%/,:,%*,%?,%",%<,%>,%|]','')
+    item.title = item.title:gsub('[\\,%/,:,%*,%?,%",%<,%>,%|]','')
```

Comment:
Let's replace invalid chars with `_` for now then. I think what @Frenzie proposed should work in practice. If getFilesystemType returns `nil`, then just assume there is no restriction on filenames.

Some minor fixes for the snipet:

```lua
function util.getFilesystemType(path)
    local type
    mounts = io.open("/proc/mounts", "r")
    if not mounts then return nil end
    while true do
        local line, mount
        line = mounts:read()
        if line == nil then
            break
        end
        mount = line:split(" ")
        -- FIXME: do a prefix compare. this should pass as long as mount[2] is a prefix of path
        if (mount[2] == path) then
            type = mount[3]
            break
        end
    end
    mounts:close()
    return type
end
```